worker/uniter/remotestate/watcher.go

```diff
@@ -754,7 +768,12 @@ func (w *RemoteStateWatcher) watchRelationUnits(
 		for unit, settings := range change.Changed {
 			relationSnapshot.Members[unit] = settings.Version
 		}
+		for app, settingsVersion := range change.AppChanged {
+			relationSnapshot.ApplicationMembers[app] = settingsVersion
+		}
 	}
+	// Wrap the Changes() with the relationId so we can process all changes
```

Comment:
I was getting very confused between `watcher.RelationUnitsWatcher` and `relationUnitsWatcher` here - maybe the latter could be renamed to highlight the wrapping behaviour more?

Reply:
renamed to "wrappedRelationUnitsWatcher" and newRelation.. => wrapRelation...